library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java

```diff
@@ -135,6 +135,25 @@ private static String getDiagnosticInfoV21(Throwable cause) {
    */
   private static final int RECONFIGURATION_STATE_QUEUE_PENDING = 2;
 
+  /**
+   * No reinit is needed
+   */
+  private static final int REINIT_STATE_NONE = 0;
+  /**
+   * The input format has just changed. Signal an end of stream to the codec so that we can
+   * retrieve the very last decoded samples from the previous format.
+   */
+  private static final int REINIT_STATE_SIGNAL_END_OF_STREAM = 1;
+  /**
+   * The end of stream has been sent, wait for the codec to acknowledge it.
+   */
+  private static final int REINIT_STATE_WAIT_END_OF_STREAM = 2;
+  /**
+   * The last sample has been processed, we can safely reinit now.
+   */
+  private static final int REINIT_STATE_DO_REINIT_NOW = 3;
```

Comment:
I don't really understand  the need for this state. Why don't you just do the reinit directly at the points where you set the state to this value, and set back to REINIT_STATE_NONE?